Mathlib/CategoryTheory/Localization/LocalizerMorphism.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2023 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Localization.Equivalence
+
+/-!
+# Morphisms of localizers
+
+A morphism of localizers consists of a functor `F : C₁ ⥤ C₂` between
+two categories equipped with morphism properties `W₁` and `W₂` such
+that `F` sends morphisms in `W₁` to morphisms in `W₂`.
+
+If `Φ : LocalizerMorphism W₁ W₂`, we introduce the condition
+`Φ.IsLocalizedEquivalence` which expresses that the induced functor
+on the localized categories is a equivalence. This condition is
+independent of the choice of the localized categories.
+
+## References
+* [Bruno Kahn and Georges Maltsiniotis, *Structures de dérivabilité*][KahnMaltsiniotis2008]
+
+-/
+
+universe v₁ v₂ v₃ v₄ v₄' v₅ v₅' v₆ u₁ u₂ u₃ u₄ u₄' u₅ u₅' u₆
+
+namespace CategoryTheory
+
+open Localization
+
+variable {C₁ : Type u₁} {C₂ : Type u₂} {C₃ : Type u₃}
+  {D₁ : Type u₄} {D₂ : Type u₅} {D₃ : Type u₆}
+  [Category.{v₁} C₁] [Category.{v₂} C₂] [Category.{v₃} C₃]
+  [Category.{v₄} D₁] [Category.{v₅} D₂] [Category.{v₆} D₂]
+  (W₁ : MorphismProperty C₁) (W₂ : MorphismProperty C₂) (W₃ : MorphismProperty C₃)
+
+/-- If `W₁ : MorphismProperty C₁` and `W₂ : MorphismProperty C₂`, a `LocalizerMorphism W₁ W₂`
+is the datum of a functor `C₁ ⥤ C₂` which sends morphisms in `W₁` to morphisms in `W₂` -/
+structure LocalizerMorphism where
+  /-- a functor between the two categories -/
+  functor : C₁ ⥤ C₂
+  /-- the functor is compatible with the `MorphismProperty` -/
+  map : W₁ ⊆ W₂.inverseImage functor
+
+namespace LocalizerMorphism
+
+/-- The identity functor as a morphism of localizers. -/
+@[simps]
+def id : LocalizerMorphism W₁ W₁ where
+  functor := 𝟭 C₁
+  map _ _ _ hf := hf
+
+variable {W₁ W₂ W₃}
+
+/-- The composition of two localizers morphisms. -/
+@[simps]
+def comp (Φ : LocalizerMorphism W₁ W₂) (Ψ : LocalizerMorphism W₂ W₃) :
+    LocalizerMorphism W₁ W₃ where
+  functor := Φ.functor ⋙ Ψ.functor
+  map _ _ _ hf := Ψ.map _ (Φ.map _ hf)
+
+variable (Φ : LocalizerMorphism W₁ W₂) (L₁ : C₁ ⥤ D₁) [L₁.IsLocalization W₁]
+  (L₂ : C₂ ⥤ D₂) [L₂.IsLocalization W₂]
+
+lemma inverts : W₁.IsInvertedBy (Φ.functor ⋙ L₂) :=
+  fun _ _ _ hf => Localization.inverts L₂ W₂ _ (Φ.map _ hf)
+
+/-- When `Φ : LocalizerMorphism W₁ W₂` and that `L₁` and `L₂` are localization functors
+for `W₁` and `W₂`, then `Φ.localizedFunctor L₁ L₂` is the induced functor on the
+localized categories. --/
+noncomputable def localizedFunctor : D₁ ⥤ D₂ :=
```

Comment:
Should this def be mentioned in the module docstring at the top of the file?

Reply:
Thanks for the suggestion!